src/components/catalog/search/CatalogSearch.tsx

```diff
@@ -62,343 +41,104 @@ class CatalogSearch extends React.Component<
 > {
   constructor(props: CatalogSearchProps | Readonly<CatalogSearchProps>) {
     super(props);
+    this.state = {
+      showSearchDialog: false,
+      isSmallDisplay: this.isSmallDisplay(),
+    };
   }
 
-  onChangeKeyCount(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardKeyCountType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_KEY_COUNT_TYPE);
-  }
-
-  onChangeSplitType(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardSplitType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_SPLIT_TYPE);
-  }
-
-  onChangeStaggeredType(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardStaggeredType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_STAGGERED_TYPE);
-  }
-
-  onChangeLedType(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardLedType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_LED_TYPE);
-  }
-
-  onChangeKeySwitchType(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardKeySwitchType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_KEY_SWITCH_TYPE);
-  }
-
-  onChangeHotswapType(
-    event: React.ChangeEvent<{
-      name?: string | undefined;
-      value: unknown;
-    }>
-  ) {
-    const value = event.target.value as
-      | IKeyboardHotswapType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_HOTSWAP_TYPE);
+  private isSmallDisplay() {
+    // eslint-disable-next-line no-undef
+    const width = window.innerWidth;
+    return width < 600;
   }
```

Comment:
@adamrocker Should we define this function as a utility function?

Reply:
Could be nice for our future. I cut this method out as a utility function.